api/explorer/api/test/test_ingest.py

```diff
@@ -0,0 +1,199 @@
+from collections import defaultdict
+from typing import Any, Callable, Mapping, cast
+from gremlin_python.driver import serializer
+from gremlin_python.driver.driver_remote_connection import DriverRemoteConnection
+from gremlin_python.process.anonymous_traversal import traversal
+
+import pytest
+
+from explorer.api import graph
+from explorer.core import model
+
+
+#######################################################################################
+# Mock Data
+#######################################################################################
+
+dummy_pkg_d = model.Package(
+    name="D",
+    output_paths={model.OutputPathName.OUT: "/D"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="D", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[],
+)
+
+
+dummy_pkg_c = model.Package(
+    name="C",
+    output_paths={model.OutputPathName.OUT: "/C"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="C",
+        version="1.0",
+        broken=False,
+        license="MIT",
+    ),
+    build_inputs=[],
+)
+
+dummy_pkg_b = model.Package(
+    name="B",
+    output_paths={model.OutputPathName.OUT: "/B"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="B", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_c,
+        )
+    ],
+)
+
+dummy_pkg_a = model.Package(
+    name="A",
+    output_paths={model.OutputPathName.OUT: "/A"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="A", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_b,
+        ),
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_d,
+        ),
+    ],
+)
+
+dummy_nix_graph = model.NixGraph(packages=[dummy_pkg_a])
+
+#######################################################################################
+# Fixture(s)
+#######################################################################################
+@pytest.fixture
+def graph_connection():
+    conn = DriverRemoteConnection(
+        "ws://localhost:8182/gremlin",
+        "g",
+        message_serializer=serializer.GraphSONMessageSerializer(),
+    )
+    g = traversal().with_remote(conn)
+    yield conn
+    # Dropping all data from graph on teardown
+    g.V().drop().iterate()
+    g.E().drop().iterate()
+    conn.close()
+
+
+#######################################################################################
+# Tests
+#######################################################################################
+
+
+def test_unit_core_to_graph_model():
+    from explorer.api.ingest import core_to_graph_model
```

Comment:
At some point in the past there was a style recommendation to only import the module which corresponds to the test suite in the scope of the test functions. IIRC this is to isolate errors which happen on import so that they occur during test evaluation rather than when pytest parses the module. This might be outdated information though. For other imports which are not the focus of the test module, the expectation is that they can be imported successfully just like other 3rd party dependencies.

Reply:
Sounds reasonable 😄